server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -1874,13 +1874,13 @@ private List<File> buildEngine() throws ExtenderException {
                 resourceFile = buildWin32Resources(mergedAppContext);
             }
 
-
             Map<String, Object> podAppContext = new HashMap<>();
             if (resolvedPods != null) {
                 podAppContext.put("frameworks", resolvedPods.getAllPodFrameworks(platform));
                 podAppContext.put("weakFrameworks", resolvedPods.getAllPodWeakFrameworks(platform));
                 podAppContext.put("libs", resolvedPods.getAllPodLibs(platform));
                 podAppContext.put("linkFlags", resolvedPods.getAllPodLinkFlags(platform));
+                podAppContext.put("osMinVersion", resolvedPods.platformMinVersion);
```

Comment:
I'm thinking it could work like this:

build_input.yml (abbrieviated):
```
  x86_64-osx:
    env:
        MACOS_MIN_VERSION:      "{{env.MACOS_MIN_VERSION}}"
    context:
        osMinVersion: "{{env.MACOS_MIN_VERSION}}"
        flags:          [,,, "-mmacosx-version-min={{osMinVersion}}" ,,,]
        linkFlags:      [,,, "-mmacosx-version-min={{osMinVersion}}" ,,,]
        cxxLinkShFlags: [,,, "-mmacosx-version-min={{osMinVersion}}" ,,,]

    compileCmd: 'clang++ ... {{#flags}}{{{.}}} {{/flags}} ... '
```

This should allow users (in their app manifest/extension to override `osMinVersion`, just as you can override the version with your required pod-version.

The actual `-mmacosx-version-min={{osMinVersion}}` will be resolved just before the compile command is invoked.

I think it's worth testing, before committing to this technical debt, and we have to resolve it later (our goal is ofc to do as little fixup code in the extender server as possible)

Reply:
> Could be good to link the PR (I couldn't find it?)

PR: https://github.com/defold/defold/tree/dev-expose-min-os-version-for-extender-builds

Reply:
The PR has been merged.